### PR TITLE
Fix some cyclic imports & reduce const enum usage

### PR DESCRIPTION
**Pull request by dsehnal**
# Description

Fix some cyclic imports and reduce the use of const enums. This should make it easier to use the library with the `isolatedModules: true` TS config.

## Actions

- [ ] Added description of changes to the `[Unreleased]` section of `CHANGELOG.md`
- [ ] Updated headers of modified files
- [ ] Added my name to `package.json`'s `contributors`

### CODE REVIEW

File: src/mol-model/structure/structure/unit.ts [hunk 1]
```diff
@@ -36,6 +36,9 @@ type Unit = Unit.Atomic | Unit.Spheres | Unit.Gaussians
 namespace Unit {
     export const enum Kind { Atomic, Spheres, Gaussians }
 
+    // To use with isolatedModules
```
Comment (arose) [inline src/mol-model/structure/structure/unit.ts:39]: I like this pattern

File: src/mol-model-props/computed/interactions/common.ts [hunk 2]
```diff
@@ -158,7 +158,7 @@ export function interactionTypeLabel(type: InteractionType): string {
     }
 }
 
-export const enum FeatureType {
+export enum FeatureType {
```
Comment (arose) [inline src/mol-model-props/computed/interactions/common.ts:161]: These are accessed fairly often, maybe worthwhile to use the `const enum FeatureType` and `enum FeatureTypes` pattern?

File: src/mol-model-props/computed/accessible-surface-area/shrake-rupley.ts [hunk 3]
```diff
@@ -89,12 +89,19 @@ namespace AccessibleSurfaceArea {
...
     }
 
-    export const enum Flag {
+    export const enum Flags {
         NA = 0x0,
         Buried = 0x1,
         Accessible = 0x2
     }
 
+    export const Flag = {
```
Comment (arose) [inline src/mol-model-props/computed/accessible-surface-area/shrake-rupley.ts:98]: For this pattern, shall we always define the `enum` in term of the `const enum`? Also we should be consistent with the use of the plural for either the `enum`or the `const enum`.

File: src/mol-data/db/column.ts [hunk 4]
```diff
@@ -93,20 +93,35 @@ namespace Column {
...
         Present = 0,
-        /** Expressed in CIF as `.` */
+        /** Expressed in CIF as `.` (= 1) */
         NotPresent = 1,
-        /** Expressed in CIF as `?` */
+        /** Expressed in CIF as `?` (= 2) */
         Unknown = 2
     }
 
+    export const ValueKind = {
```
Comment (arose) [inline src/mol-data/db/column.ts:108]: Define in terms of the `const enum`?

**Review by arose — COMMENTED** [review]
Nice, thanks for tackling this. Shall we add a lint rule to allow `const enums` only with a comment?

File: src/mol-theme/color/_categories.ts [hunk 5]
```diff
@@ -0,0 +1,14 @@
+/**
```
[the remark this answers is not in the document]
Comment (dsehnal) [reply src/mol-theme/color/_categories.ts:1]: Wanted it to be on top, but yeah, can rename it.

**Comment by dsehnal** [comment]
> Shall we add a lint rule to allow const enums only with a comment?

Perhaps. Know how to do this?

**Pull request merged**